libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2508,29 +2507,56 @@ string YulUtilFunctions::revertReasonIfDebug(RevertStrings revertStrings, string
 				mstore(add(reasonPos, <offset>), <wordValue>)
 			</word>
 			revert(0, add(reasonPos, <end>))
-		})");
-		templ("sig", (u256(util::FixedHash<4>::Arith(util::FixedHash<4>(util::keccak256("Error(string)")))) << (256 - 32)).str());
+		<!debugAndMessage>
+			revert(0, 0)
+		</debugAndMessage>
+	})");
+	bool debugAndMessage = _revertStrings >= RevertStrings::Debug && !_message.empty();
+	templ("debugAndMessage", debugAndMessage);
+	if (debugAndMessage)
+	{
+		templ("sig", (u256(util::FixedHash<4>::Arith(util::FixedHash<4>(util::keccak256("Error(string)"))))
+				<< (256 - 32)).str());
```

Comment:
I think that splitting the line in the middle of an expression is bad for readability. If I had to, I'd split it like this:
```suggestion
		templ(
			"sig",
			(
				u256(util::FixedHash<4>::Arith(util::FixedHash<4>(util::keccak256("Error(string)")))) <<
				(256 - 32)
			).str()
		);
```
That's pretty long-winded though, so how about this?
```suggestion
		auto sig32 = util::FixedHash<4>::Arith(util::FixedHash<4>(util::keccak256("Error(string)")));
		templ("sig", (u256(sig32) << (256 - 32)).str());
```